apps/web/components/booking/BookingListItem.tsx

```diff
@@ -115,6 +116,38 @@ function BookingListItem(booking: BookingItemProps) {
     return booking.seatsReferences[0].referenceUid;
   };
 
+  // Check user's availability/slots for the current booking request,
+  const isSlotAvailable = () => {
+    const resp = trpc.viewer.public.slots.getSchedule.useQuery(
+      {
```

Comment:
is there any other way to check this. this API is sometimes very slow on production and can slow this page also if user has many unconfirmed bookings